tests/kafkatest/tests/verifiable_share_consumer_test.py

```diff
@@ -0,0 +1,106 @@
+from kafkatest.services.kafka import TopicPartition
+
+class VerifiableShareConsumerTest(KafkaTest):
+    PRODUCER_REQUEST_TIMEOUT_SEC = 30
+
+    def __init__(self, test_context, num_consumers=1, num_producers=0, **kwargs):
+        super(VerifiableShareConsumerTest, self).__init__(test_context, **kwargs)
+        self.num_consumers = num_consumers
+        self.num_producers = num_producers
+
+    def _all_partitions(self, topic, num_partitions):
+        partitions = set()
+        for i in range(num_partitions):
+            partitions.add(TopicPartition(topic=topic, partition=i))
+        return partitions
+
+    def _partitions(self, assignment):
+        partitions = []
+        for parts in assignment.values():
+            partitions += parts
+        return partitions
+
+    def valid_assignment(self, topic, num_partitions, assignment):
+        all_partitions = self._all_partitions(topic, num_partitions)
+        partitions = self._partitions(assignment)
+        return len(partitions) == num_partitions and set(partitions) == all_partitions
+
+    def min_cluster_size(self):
+        """Override this since we're adding services outside of the constructor"""
+        return super(VerifiableShareConsumerTest, self).min_cluster_size() + self.num_consumers + self.num_producers
+
+    def setup_share_group(self, topic, acknowledgement_mode="auto", group_id="test_group_id", offset_reset_strategy="", **kwargs):
+        return VerifiableShareConsumer(self.test_context, self.num_consumers, self.kafka,
+                                  topic, group_id, acknowledgement_mode=acknowledgement_mode,
+                                  offset_reset_strategy=offset_reset_strategy, log_level="TRACE", **kwargs)
+
+    def setup_producer(self, topic, max_messages=-1, throughput=500):
+        return VerifiableProducer(self.test_context, self.num_producers, self.kafka, topic,
+                                  max_messages=max_messages, throughput=throughput,
+                                  request_timeout_sec=self.PRODUCER_REQUEST_TIMEOUT_SEC,
+                                  log_level="DEBUG")
+
+    def await_produced_messages(self, producer, min_messages=1000, timeout_sec=10):
+        current_acked = producer.num_acked
+        wait_until(lambda: producer.num_acked >= current_acked + min_messages, timeout_sec=timeout_sec,
+                   err_msg="Timeout awaiting messages to be produced and acked")
+
+    def await_consumed_messages(self, consumer, min_messages=1, timeout_sec=10, total=False):
+        current_total = 0
+        if total is False:
+            current_total = consumer.total_consumed()
+        wait_until(lambda: consumer.total_consumed() >= current_total + min_messages,
+                   timeout_sec=timeout_sec,
+                   err_msg="Timed out waiting for consumption")
+
+    def await_consumed_messages_by_a_consumer(self, consumer, node, min_messages=1, timeout_sec=10, total=False):
```

Comment:
Can you explain the purpose of `await_consumed_messages_by_a_consumer` as opposed to `await_consumed_messages`? This is done in a different way than the regular `consumer_test.py` and `verifiable_consumer_test.py` and I wonder what the motivation is for doing it differently.